### ISSUE TITLE
npm package install failed on termux android: Unhandled rejection Error

### ISSUE THREAD

**Issue by whao** [npm/npm#17568]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [x] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

#### What's going wrong?
```
#  @ localhost in ~ [0:10:10] C:1
$ npm -g install typescript
Unhandled rejection Error: EACCES: permission denied, link '/data/data/com.termux/files/home/.npm/_cacache/tmp/73a10869' -> '/data/data/com.termux/files/home/.npm/_cacache/content-v2/sha512/14/fa/03e3ff06ab7374fd1397c38b8ff0318fb86021f07d2007876bfbeebd1e64fe495777396f98983eb59867a41530ffab4799578efcacb911a49fbe53dd4203'
    at Error (native)

npm ERR! cb() never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://github.com/npm/npm/issues>

npm ERR! A complete log of this run can be found in:
npm ERR!     /data/data/com.termux/files/home/.npm/_logs/2017-07-02T16_10_35_407Z-debug.log
```
cat log
```
#  @ localhost in ~ [0:23:49] C:126
$ cat /data/data/com.termux/files/home/.npm/_logs/2017-07-02T16_10_35_407Z-debug.log
0 info it worked if it ends with ok
1 verbose cli [ '/data/data/com.termux/files/usr/bin/node',
1 verbose cli   '/data/data/com.termux/files/usr/bin/npm',
1 verbose cli   '-g',
1 verbose cli   'install',
1 verbose cli   'typescript' ]
2 info using npm@5.0.4
3 info using node@v6.11.0
4 verbose npm-session d357418d42fb80e5
5 silly install loadCurrentTree
6 silly install readGlobalPackageData
7 error cb() never called!
8 error This is an error with npm itself. Please report this error at:
9 error <https://github.com/npm/npm/issues>
```
#### How can the CLI team reproduce the problem?
Download termux on Android, install nodejs by `pkg install nodejs`. Update npm to the latest version by `npm -g instsll npm`. Then `npm -g install` any package will cause this issue.

### supporting information:

 - `npm -v` prints: `5.0.4`
 - `node -v` prints: `v6.11.0`
 - `npm config get registry` prints: `https://registry.npmjs.org/`
 - Windows, OS X/macOS, or Linux?: Termux on Android
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by ciscoo** [comment]
I don't think this is an issue with npm itself. You're clearly getting a permissions error: `Unhandled rejection Error: EACCES: permission denied (...)`. Since you're running this on Android it's likely your permissions aren't correct. Most likely npm is attempting to write to a location in which it does not have permissions to do so.

**Comment by christianrondeau** [comment]
I think this is a duplicate of https://github.com/npm/npm/issues/17480 (we should close one or the other). It's not a permission issue, at least not as far as I know. There's something fishy going on there. I can manually create and symlink stuff in both the source and target folder, and there's a single non-root user in Termux. I'm not sure the [the referenced issue in termux](https://github.com/termux/termux-app/issues/351) is actually related, because if you look at the paths from the error message (and from my related issue), they are in the `/home` folder.

**Comment by abraxas86** [comment]
I'm not familiar with NPM, but I'm having the exact same issue with another Python script that I downloaded from GitHub and compiled via Termux.  

I'm assuming the issue is probably more related to either Termux, or the Python package in Termux's repo that is the issue.
